src/System.Private.CoreLib/shared/System/Guid.cs

```diff
@@ -656,6 +669,117 @@ private static bool TryParseExactX(ReadOnlySpan<char> guidString, ref GuidResult
             return true;
         }
 
+        private static bool TryParseHexD(ReadOnlySpan<char> value, out uint result)
+        {
+            Debug.Assert(value.Length == 4, "Expects exact 4 (guaranteed, because called only from TryParseExactD)");
+
+            ReadOnlySpan<byte> charToHexLookup = Number.CharToHexLookup;
+            uint numValue;
+            result = 0;
+
+            // First char
+            int num = value[0];
+            
+            // Can start only with hex digit or "+" (43 < charToHexLookup.Length); "0x" means '0' for num
+            if ((uint)num >= (uint)charToHexLookup.Length)
+                goto FalseExit;
+
+            if ((numValue = charToHexLookup[num]) == 0xFF)
+                goto FirstCharNotHex;
```

Comment:
This looks very similar to https://github.com/dotnet/coreclr/pull/21442/files#diff-f9c6f1fc0b086f75b048b734798233c7L667. Can this be more unified and/or shared?